clang/lib/CodeGen/CGExpr.cpp

```diff
@@ -2440,6 +2443,35 @@ RValue CodeGenFunction::EmitLoadOfLValue(LValue LV, SourceLocation Loc) {
         Builder.CreateLoad(LV.getMatrixAddress(), LV.isVolatileQualified());
     return RValue::get(Builder.CreateExtractElement(Load, Idx, "matrixext"));
   }
+  if (LV.isMatrixRow()) {
+    QualType MatTy = LV.getType();
+    const ConstantMatrixType *MT = MatTy->castAs<ConstantMatrixType>();
+
+    unsigned NumRows = MT->getNumRows();
+    unsigned NumCols = MT->getNumColumns();
+
+    llvm::Value *MatrixVec = EmitLoadOfScalar(LV, Loc);
+
+    llvm::Value *Row = LV.getMatrixRowIdx();
+    llvm::Value *Result =
+        llvm::PoisonValue::get(ConvertType(LV.getType())); // <NumCols x T>
+
+    llvm::MatrixBuilder MB(Builder);
+
+    for (unsigned Col = 0; Col < NumCols; ++Col) {
+      llvm::Value *ColIdx = llvm::ConstantInt::get(Row->getType(), Col);
+
+      llvm::Value *EltIndex = MB.CreateIndex(Row, ColIdx, NumRows);
+
+      llvm::Value *Elt = Builder.CreateExtractElement(MatrixVec, EltIndex);
+
+      llvm::Value *Lane = llvm::ConstantInt::get(Builder.getInt32Ty(), Col);
+
+      Result = Builder.CreateInsertElement(Result, Elt, Lane);
```

Comment:
```suggestion
      llvm::Value *ColIdx = llvm::ConstantInt::get(Row->getType(), Col);
      llvm::Value *EltIndex = MB.CreateIndex(Row, ColIdx, NumRows);
      llvm::Value *Elt = Builder.CreateExtractElement(MatrixVec, EltIndex);
      llvm::Value *Lane = llvm::ConstantInt::get(Builder.getInt32Ty(), Col);
      Result = Builder.CreateInsertElement(Result, Elt, Lane);
```

Could you please reduce the number of empty new lines? Same around line 2713.